libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -0,0 +1,346 @@
+                {
+                    recognizerResult.Intents.Add(NoneIntent, new IntentScore() { Score = 1.0 });
+                }
+                else
+                {
+                    // add top score
+                    AddResultAsIntent(recognizerResult, results[0]);
+
+                    // Disambiguate if configured
+                    if (detectAmbiguity)
+                    {
+                        var thresholdScore = DisambiguationScoreThreshold.GetValue(dc.State);
+                        var classifyingScore = Math.Round(topScore, 2) - Math.Round(thresholdScore, 2);
+                        var ambigiousResults = results.Where(item => item.Score >= classifyingScore).ToList();
+
+                        if (ambigiousResults.Count > 1)
+                        {
+                            // create a RecognizerResult for each ambigious result.
+                            var recognizerResults = ambigiousResults.Select(result => new RecognizerResult()
+                            {
+                                Text = text,
+                                AlteredText = result.ClosestText,
+                                Entities = recognizerResult.Entities,
+                                Properties = recognizerResult.Properties,
+                                Intents = new Dictionary<string, IntentScore>()
+                                {
+                                    { result.Label.Name, new IntentScore() { Score = result.Score } }
+                                },
+                            });
+
+                            // replace RecognizerResult with ChooseIntent => Amibgious recognizerResults as candidates. 
+                            recognizerResult = CreateChooseIntentResult(recognizerResults.ToDictionary(result => this.Id, result => result));
+                        }
+                    }
+                }
+            }
+            else
+            {
+                // Return 'None' if no intent matched.
+                recognizerResult.Intents.Add(NoneIntent, new IntentScore() { Score = 1.0 });
+            }
+
+            await dc.Context.TraceActivityAsync(nameof(OrchestratorRecognizer), JObject.FromObject(recognizerResult), nameof(OrchestratorRecognizer), "Orchestrator Recognition ", cancellationToken).ConfigureAwait(false);
+            TrackRecognizerResult(dc, nameof(OrchestratorRecognizer), FillRecognizerResultTelemetryProperties(recognizerResult, telemetryProperties), telemetryMetrics);
+
+            return recognizerResult;
+        }
+
+        private void AddResultAsIntent(RecognizerResult recognizerResult, Result result)
+        {
+            recognizerResult.Intents.Add(result.Label.Name, new IntentScore()
+            {
+                Score = result.Score
+            });
+        }
+
+        private async Task<RecognizerResult> RecognizeEntitiesAsync(DialogContext dialogContext, Schema.Activity activity, RecognizerResult recognizerResult)
```

Comment:
where is this called from? (its private I can't find where it is referenced)

nice t see this just delegates to Recognizer Set - but it would be good to know that i can call Recognizer without invoking this entity stuff if i so wished.

Reply:
Fixed. Tom removed this via one of his commits :)